plugin/src/main/java/com/yelp/codegen/KotlinGenerator.kt

```diff
@@ -205,6 +205,8 @@ open class KotlinGenerator : SharedCodegen() {
 
     @VisibleForTesting
     internal fun addRequiredImports(codegenModel: CodegenModel) {
+        // Import JsonClass annotation to enable Adapters generations via Kotlin Annotation Processor
+        codegenModel.imports.add("com.squareup.moshi.JsonClass")
```

Comment:
Can be moved inside the `if` block below

Reply:
I totally missed the other PRs that were opened ... sorry :(

I'll check if we can move the import into the if statement, I'll let you know soon about the progress.

Reply:
`JsonClass` should be imported in all the model files, except if the model is actually a type alias.
Going to address this edge-case